dev/release-notes.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+#
+# Usage: ./release-notes.py YYYY-MM-DD
+#
+
+import sys
+import json
+import os.path
+from github import Github
+from datetime import datetime
+
+def usage():
+    print("Usage: ./release-notes.py YYYY-MM-DD")
+
+def get_prs(repo,startdate):
+    """Retrieves data for PRs matching selection criteria and puts them in a dictionary,
+       which is then saved in a json file, and also returned for immediate use."""
+    prs = {}
+    all_pulls = repo.get_pulls(state="closed", sort="created", direction="desc", base="master")
+    # We need to run this over the whole list of PRs. Sorting by creation date descending
+    # is not really helping - could be that some very old PRs are being merged.
+    for pr in all_pulls:
+        print(pr.number, end=" ")
+        # flush stdout immediately, to see progress indicator
+        sys.stdout.flush()
+        if pr.merged:
+            if pr.closed_at > datetime.fromisoformat(startdate):
+                labs = [lab.name for lab in list(pr.get_labels())]
+                prs[pr.number] = { "title" : pr.title,
+                                    "closed_at" : pr.closed_at.isoformat(),
+                                    "labels" : labs }
+#                 if len(prs)>5: # for quick testing (maybe later have an optional argument)
+#                     break
+    print("\n")
+    with open("prscache.json", "w", encoding="utf-8") as f:
+        json.dump(prs, f, ensure_ascii=False, indent=4)
+    return prs    
+
+def changes_overview(prs,startdate):
+    """Writes files with information for release notes."""
+
+    #TODO: using cached data, check that the starting date is the same
+    # Also save the date when cache was saved (warning if old?)
```

Comment:
I am thinking that the default behaviour should ignore the cache. One would generate release notes, have a pass over them, find improvements for labels and PR titles, make changes on GitHub and then rerun it to retrieve new data from GitHub. So this will be an option for testing the output part, not the default.